libraries/FreeRTOS/src/variantHooks.cpp

```diff
@@ -70,8 +70,14 @@ extern "C" {
         xSemaphoreGive(mtx);
     }
 
-    void __freertos_mutex_give_from_isr(SemaphoreHandle_t mtx) {
-        xSemaphoreGiveFromISR(mtx, NULL);
+    void __freertos_mutex_give_from_isr(SemaphoreHandle_t mtx, BaseType_t* pxHigherPriorityTaskWoken) {
+        if (!pxHigherPriorityTaskWoken) {
```

Comment:
The GiveFromISR might also set the flag, so I think the `if` is not needed or correct, no?

Given the description, should this whole function be
````
            xSemaphoreGiveFromISR(mtx, pxHigherPriorityTaskWoken);
            portYIELD_FROM_ISR(pxHigherPriorityTaskWoken);
````
?

Reply:
I implemented it that way because I am not sure how `portYIELD_FROM_ISR(NULL);` is working.

Will check.

Reply:
Maybe it's as simple as this?  You pass in NULL, it starts at FALSE, OTW it starts at whatever the earlier call was.
````
void __freertos_mutex_give_from_isr(SemaphoreHandle_t mtx, BaseType_t* pxHigherPriorityTaskWoken) {
    BaseType_t hiprio = pxHigherPriorityTaskWoken ? *pxHigherPriorityTaskWoken : pdFALSE;
    xSemaphoreGiveFromISR(mtx, &hiprio);
    portYIELD_FROM_ISR(&hiPrio);
}
````